cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,613 @@
+from widgetastic_patternfly import TextInput
+
+from . import MigrationView
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.log import logger
+from cfme.utils.version import Version
+from cfme.utils.version import VersionPicker
+from cfme.utils.wait import wait_for
+from widgetastic_manageiq import HiddenFileInput
+from widgetastic_manageiq import InfraMappingList
+from widgetastic_manageiq import MigrationDashboardStatusCard
+from widgetastic_manageiq import MigrationDropdown
+from widgetastic_manageiq import MigrationPlanRequestDetailsList
+from widgetastic_manageiq import MigrationPlansList
+from widgetastic_manageiq import MigrationProgressBar
+from widgetastic_manageiq import RadioGroup
+from widgetastic_manageiq import Table
+from widgetastic_manageiq import V2VPaginatorPane
+
+
+class DashboardCards(View):
+    not_started_plans = MigrationDashboardStatusCard("Not Started")
+    in_progress_plans = MigrationDashboardStatusCard("In Progress")
+    completed_plans = MigrationDashboardStatusCard("Complete")
+    archived_plans = VersionPicker(
+        {
+            Version.lowest(): MigrationDashboardStatusCard("Infrastructure Mappings"),
+            "5.10": MigrationDashboardStatusCard("Archived"),
+        }
+    )
+
+    def read(self):
+        return [c.read().get("name") for c in self.sub_widgets if c.is_active]
+
+
+class MigrationPlanView(MigrationView):
+    dashboard_cards = View.nested(DashboardCards)
+    plans_list = ConditionalSwitchableView(reference="dashboard_cards")
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    title = Text('.//div[contains(@class, "pull-left")]//h3')
+    create_migration_plan = Text(locator='(//a|//button)[text()="Create Migration Plan"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    sort_type_dropdown = SelectorDropdown("id", "sortTypeMenu")
+    sort_direction = Text(locator=".//span[contains(@class,'sort-direction')]")
+    progress_card = MigrationProgressBar(locator=".//div[3]/div/div[3]/div[3]/div/div")
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    # 5.9 fields
+    migr_dropdown = MigrationDropdown(text="Not Started Plans")
+    infra_mapping_list = InfraMappingList("infra-mappings-list-view")
+    create_infra_mapping = Text(locator='(//a|//button)[text()="Create Infrastructure Mapping"]')
+
+    @property
+    def in_migration_plan(self):
+        return (
+            len(self.browser.elements('.//div[contains(@class,"card-pf")]')) > 0
+            and len(self.browser.elements(PFIcon.icons.WARNING)) < 1
+        )
```

Comment:
Consider checking the breadcrumb here?

Reply:
Moved migrationView to this class.
This is inheriting MigrationView which will check navigation menu

Reply:
Well this isn't calling the `is_displayed` in `MigrationView`, you would have to explicitly call it. 

How this `is_displayed` is currently written, any view that has more than 1 card will return `True`. I think it be better to check the breadcrumb and the expected widgets on the view